arm/ui/utils.py

```diff
@@ -79,42 +83,176 @@ def check_db_version(install_path, db_file):
 
         if not os.path.isfile(db_file):
             app.logger.debug("Can't create database file.  This could be a permissions issue.  Exiting...")
+        else:
+            # Only run the below if the db exists
+            # Check to see if db is at current revision
+            head_revision = script.get_current_head()
+            app.logger.debug("Alembic Head is: " + head_revision)
 
-    # check to see if db is at current revision
-    head_revision = script.get_current_head()
-    app.logger.debug("Head is: " + head_revision)
-
-    conn = sqlite3.connect(db_file)
-    c = conn.cursor()
+            conn = sqlite3.connect(db_file)
+            c = conn.cursor()
 
-    c.execute('SELECT version_num FROM alembic_version')
-    db_version = c.fetchone()[0]
-    app.logger.debug(f"Database version is: {db_version}")
-    if head_revision == db_version:
-        app.logger.info("Database is up to date")
-    else:
-        app.logger.info(
-            f"Database out of date. Head is {head_revision} and database is {db_version}.  Upgrading database...")
-        with app.app_context():
-            unique_stamp = round(time() * 100)
-            app.logger.info(f"Backing up database '{db_file}' to '{db_file}{unique_stamp}'.")
-            shutil.copy(db_file, db_file + "_" + str(unique_stamp))
-            flask_migrate.upgrade(mig_dir)
-        app.logger.info("Upgrade complete.  Validating version level...")
-
-        c.execute("SELECT version_num FROM alembic_version")
+        c.execute('SELECT version_num FROM alembic_version')
         db_version = c.fetchone()[0]
         app.logger.debug(f"Database version is: {db_version}")
         if head_revision == db_version:
-            app.logger.info("Database is now up to date")
+            app.logger.info("Database is up to date")
+        else:
+            app.logger.info(
+                f"Database out of date. Head is {head_revision} and database is {db_version}.  Upgrading database...")
+            with app.app_context():
+                unique_stamp = round(time() * 100)
+                app.logger.info(f"Backing up database '{db_file}' to '{db_file}{unique_stamp}'.")
+                shutil.copy(db_file, db_file + "_" + str(unique_stamp))
+                flask_migrate.upgrade(mig_dir)
+            app.logger.info("Upgrade complete.  Validating version level...")
+
+            c.execute("SELECT version_num FROM alembic_version")
+            db_version = c.fetchone()[0]
+            app.logger.debug(f"Database version is: {db_version}")
+            if head_revision == db_version:
+                app.logger.info("Database is now up to date")
+            else:
+                app.logger.error(f"Database is still out of date. "
+                                 f"Head is {head_revision} and database is {db_version}.  Exiting arm.")
+
+
+def arm_alembic_get():
```

Comment:
In the future, it might be a good idea to factor these functions out into a `db_utils` file

Reply:
I am keen to move the database handler to a class and call it from the ui and ripper to avoid duplication of code. However, in developing this change was not sure on how the project was setup and the ripper code looked to hard to update without breaking everything.
How do we keep track of this? I can't add a task to the 2.7 project area